### PR TITLE
NIAD-1592: Immunization: When site is processed into to text the term text is incorrect for synonyms

### PR DESCRIPTION
**Pull request by bbell-kainos**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java [hunk 1]
```diff
@@ -35,4 +49,16 @@ public static boolean hasCode(CodeableConcept code, List<String> codeLists) {
             .map(Coding::getCode)
             .anyMatch(codeLists::contains);
     }
+
+    private static String getExtensionTermText(List<Extension> extensions) {
+        return extensions.stream()
+            .filter(extension -> extension.getUrl().equals(DISPLAY_EXTENSION_URL))
```
Comment (bartek-sarul) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java:55]: Always compare const to a variable.
```
DISPLAY_EXTENSION_URL.equals(extension.getUrl())
```

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java [hunk 2]
```diff
@@ -35,4 +49,16 @@ public static boolean hasCode(CodeableConcept code, List<String> codeLists) {
...
+            .map(Extension.class::cast)
+            .map(Extension::getExtension)
+            .flatMap(List::stream)
+            .filter(extension -> extension.getUrl().equals(DESCRIPTION_DISPLAY_URL))
```
Comment (bartek-sarul) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java:59]: The same comment as above

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

**Review by rorysproule1 — APPROVED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java (hunk 1 above)
Comment (bbell-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java]: Addressed

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java (hunk 2 above)
Comment (bbell-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/CodeableConceptMappingUtils.java]: Addressed

**Review by rorysproule1 — APPROVED** [review]

**Review by bartek-sarul — APPROVED** [review]

**Pull request merged**